internal/metamorph/processor.go

```diff
@@ -857,3 +878,25 @@ func callbackExists(callback store.Callback, data *store.Data) bool {
 	}
 	return false
 }
+
+func (p *Processor) saveTxToCache(hash *chainhash.Hash) {
+	if p.cacheStore.IsShared() {
+		_ = p.cacheStore.MapSet(CacheRegisteredTxsHash, hash.String(), []byte("1"))
```

Comment:
I think here we can set the hash directly and not as a map with a certain ttl e.g. 5min

```suggestion
p.cacheStore.Set(hash.String(), []byte("1"), 5 * time.Minute)
```

Also we should handle the error

Reply:
> When we re-broadcast the transaction here, we need to make sure that the hash is still in the cache

You're right, I added that.

---

> Also we should handle the error

You're right, I added logging the error, but I don't know whether to return an error in `ProcessTransaction` or just proceed? If the error is just some temporary hiccup, we could still proceed to announce the transaction and it will try to be added to the cache again when re-broadcasting. WDYT?

---

> I think here we can set the hash directly and not as a map with a certain ttl e.g. 5min

I didn't want to set the hash directly, in case we might need to use something like `GetAllFromMap()` from cache. Then, we wouldn't want to get everything that's in the general cache, but only things from the set. But for now, I guess we don't need it in this case.

As for TTL, I think you're right that it would be useful to clear the cache in case the transactions are not removed in the "normal usecases" for some reason.

If we want to keep it in the Map, there's a way of setting TTL with the newer version of `go-redis` library, with method `cache.HExpire()`. I can update it and use it.

Otherwise, if we don't care about map, I will just use the general cache and TTL that's already implemented. Let me know @boecklim